packages/vm/src/evm/interpreter.ts

```diff
@@ -140,13 +141,29 @@ export default class Interpreter {
    */
   async runStep(): Promise<void> {
     const opInfo = this.lookupOpInfo(this._runState.opCode)
+
+    const gas = new BN(opInfo.fee)
+    // clone the gas limit; call opcodes can add stipend,
+    // which makes it seem like the gas left increases
+    const gasLimitClone = this._eei.getGasLeft()
+
+    if (opInfo.dynamicGas) {
+      const dynamicGasHandler = dynamicGasHandlers.get(this._runState.opCode)!
+      // This function updates the gas BN in-place using `i*` methods
+      // It needs the base fee, for correct gas limit calculation for the CALL opcodes
+      await dynamicGasHandler(this._runState, gas)
+    }
+
+    // TODO: figure out if we should try/catch this (in case step event throws)
```

Comment:
do you have an example of when or how the step event would throw? it seems innocent enough with some debug and `emit`.

if it throws, in some extreme case, we would probably want the error to bubble up to the user, although we could wrap it more nicely.

Reply:
If we use a `number` instead of `BN` (so we convert the BN to a number) then it might throw if the BN does not fit in a number. What I actually meant with this TODO is to figure out what happens if any of the listeners throws. The previous behavior is that the tx then stops executing, so I guess we should keep that.

Reply:
> What I actually meant with this TODO is to figure out what happens if any of the listeners throws. The previous behavior is that the tx then stops executing, so I guess we should keep that.

ooh I see what you mean, if the consumer of the step event has a listener that throws? i thought you were maybe referring to it throwing within our own code, but it externally throwing also makes sense.

> If we use a number instead of BN (so we convert the BN to a number) then it might throw if the BN does not fit in a number.

yeah, in what kind of situation would the fee be `> 2^53 - 1` though? seems like a pretty extreme case. i only mention it because then we can merge this in without a breaking change and then update the number to BN in a future planned breaking changes release (which we can get started on as soon as london goes live on mainnet)

Reply:
I have ran into the number being too large, I am assuming this was some ethereum/tests test. It could theoretically also happen if we implement the trace specification; then if we normally would forward "all" gas, then Geth would report this amount of gas which would get forwarded, so if we have blocks with extremely large gas limits (i.e. in ethereum/tests) then it would report a number which is too large.

Reply:
circling back to this TODO, do you have any fresh thoughts on this?

`// TODO: figure out if we should try/catch this (in case step event throws)`

i figure it would be harmless to add a try/catch as an extra precaution, can even add a test for it where  a listener on the `step` event throws and expect that it doesn't stop vm execution.

Reply:
Not sure if this has been addressed. Will nevertheless merge this in here, in doubt we can add separtely along a follow-up PR.